backend/src/main/java/com/huellapositiva/domain/actions/FetchVolunteerProfileAction.java

```diff
@@ -0,0 +1,67 @@
+
+@Service
+public class FetchVolunteerProfileAction {
+
+    @Autowired
+    private JpaVolunteerRepository jpaVolunteerRepository;
+
+    /**
+     * This method fetches a Volunteer with full information (with partially credentials and location) from the DB
+     *
+     * @param volunteerEmail Email of the logged volunteer
+     */
+    public ProfileDto execute(String volunteerEmail) {
+        JpaVolunteer jpaVolunteer = jpaVolunteerRepository.findByEmailProfileInformation(volunteerEmail);
+
+        boolean jpaLocationIsNull = jpaVolunteer.getLocation() == null;
+        if (jpaLocationIsNull) {
+            return ProfileDto.builder()
+                    .name(jpaVolunteer.getCredential().getName())
+                    .surname(jpaVolunteer.getCredential().getSurname())
+                    .birthDate(jpaVolunteer.getCredential().getBirthDate() == null ?
+                        null : jpaVolunteer.getCredential().getBirthDate().format(DateTimeFormatter.ISO_LOCAL_DATE))
+                    .phoneNumber(jpaVolunteer.getCredential().getPhoneNumber())
+                    .email(jpaVolunteer.getCredential().getEmail())
+                    .province(null)
+                    .town(null)
+                    .zipCode(null)
+                    .address(null)
+                    .photo(jpaVolunteer.getPhotoUrl())
+                    .curriculumVitae(jpaVolunteer.getCurriculumVitaeUrl())
+                    .twitter(jpaVolunteer.getTwitter())
+                    .instagram(jpaVolunteer.getInstagram())
+                    .linkedin(jpaVolunteer.getLinkedin())
+                    .additionalInformation(jpaVolunteer.getAdditionalInformation())
+                    .build();
+        } else {
+            return ProfileDto.builder()
+                    .name(jpaVolunteer.getCredential().getName())
+                    .surname(jpaVolunteer.getCredential().getSurname())
+                    .birthDate(jpaVolunteer.getCredential().getBirthDate() == null ?
+                            null : jpaVolunteer.getCredential().getBirthDate().format(DateTimeFormatter.ISO_LOCAL_DATE))
+                    .phoneNumber(jpaVolunteer.getCredential().getPhoneNumber())
+                    .email(jpaVolunteer.getCredential().getEmail())
+                    .province(jpaVolunteer.getLocation().getProvince())
+                    .town(jpaVolunteer.getLocation().getTown())
+                    .zipCode(jpaVolunteer.getLocation().getZipCode())
+                    .address(jpaVolunteer.getLocation().getAddress())
+                    .photo(jpaVolunteer.getPhotoUrl())
+                    .curriculumVitae(jpaVolunteer.getCurriculumVitaeUrl())
+                    .twitter(jpaVolunteer.getTwitter())
+                    .instagram(jpaVolunteer.getInstagram())
+                    .linkedin(jpaVolunteer.getLinkedin())
+                    .additionalInformation(jpaVolunteer.getAdditionalInformation())
+                    .build();
+        }
```

Comment:
Refactor

Reply:
Fixed.

Reply:
The implementation have changed because we have changed the database. I shall to check "Resolve conversation"